src/app/pages/record/add-event/add-event.component.ts

```diff
@@ -1,14 +1,54 @@
-import { Component, Inject, NgIterable } from '@angular/core';
-import { ICategories } from '../../history/history.interface';
-import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { Component, Inject, NgIterable, OnDestroy, OnInit } from '@angular/core';
+import { ICategories, IEvents } from '../../history/history.interface';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { RecordService } from '../record.service';
+import { RecordComponent } from '../record.component';
+import { HistoryComponent } from '../../history/history.component';
 
 @Component({
   selector: 'app-add-event',
   templateUrl: './add-event.component.html',
   styleUrls: ['./add-event.component.css']
 })
-export class AddEventComponent {
+export class AddEventComponent implements OnInit, OnDestroy {
 
-  constructor(@Inject(MAT_DIALOG_DATA) public data: NgIterable<ICategories>) { }
+  public form!: FormGroup;
+
+  private sub: Subscription = new Subscription();
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: NgIterable<ICategories>,
+    public dialogRef: MatDialogRef<RecordComponent | HistoryComponent>,
+    private recordService: RecordService) { }
+
+  public ngOnInit(): void {
+    this.buildForm();
+  }
+
+  public ngOnDestroy(): void {
+    this.sub.unsubscribe();
+  }
+
+  public addEvents(): void {
+    const event: IEvents = this.form.value;
```

Comment:
why you create new variable? why we can just write `this.recordService.addEvent(this.form.value)`